int8    pos_id
float32 distance